applications/StructuralMechanicsApplication/python_scripts/convergence_criteria_factory.py

```diff
@@ -33,23 +32,23 @@ def __init__(self, convergence_criterion_parameters):
                 rotation_dofs = True
         
         # Convergence criteria if there are rotation DOFs in the problem
-        if(rotation_dofs == True):
-            if(convergence_crit == "displacement_criterion"):
+        if(rotation_dofs is True):
+            if("displacement_criterion" in convergence_crit):
                 self.mechanical_convergence_criterion = StructuralMechanicsApplication.DisplacementAndOtherDoFCriteria(D_RT, D_AT)
                 self.mechanical_convergence_criterion.SetEchoLevel(echo_level)
                 
-            elif(convergence_crit == "residual_criterion"):
+            elif("residual_criterion" in convergence_crit):
```

Comment:
What is wrong with `==`?
Now I can make it fail if I input sth like `supid_string_that_contains_residual_criterion`

Reply:
The problem is that if I use the "residual_criterion_with_adaptative_remeshing" I need to consider many exceptions instead of reusing what is already

Reply:
i think that @philbucher 's comment is fair...this does not look a good way to solve your problem